runtime/commands/default.js

```diff
@@ -740,6 +740,147 @@ Commands.ban = {
   }
 }
 
+Commands.hackban = {
+  name: 'hackban',
+  help: 'Swing the ban hammer on someone who isn\'t a member of the server!',
+  noDM: true,
+  usage: '<userid | userids> <optional reason>',
+  level: 0,
+  fn: function (msg, suffix, bot) {
+    var guildPerms = msg.author.permissionsFor(msg.guild)
+    var botPerms = bot.User.permissionsFor(msg.guild)
+
+    if (!guildPerms.General.BAN_MEMBERS) {
+      msg.reply('You do not have Ban Members permission here.')
+    } else if (!suffix) {
+      msg.channel.sendMessage('You need to provide an ID to ban!')
+    } else if (!botPerms.General.BAN_MEMBERS) {
+      msg.channel.sendMessage('I do not have Ban Members permission, sorry!')
+    } else if (msg.mentions.length > 0) {
+      msg.channel.sendMessage('You need to provide an ID to ban! Mentions aren\'t supported for hackban.')
+    } else {
+      msg.reply('Please wait...').then((m) => {
+        let banMembers = {success: [], error: []}
+        let idArray = []
+        let reasonWords = []
+        suffix.split(' ').map((id) => {
+          if (isNaN(id)) {
+            reasonWords.push(id)
+          } else {
+            idArray.push(id)
+          }
+        })
+        let reason = reasonWords.length > 0 ? reasonWords.join(' ') : 'no reason provided.'
+        idArray.map((id) => {
+          bot.Users.getREST(id).then((user) => {
+            msg.guild.ban(id, 0, `${msg.author.username}#${msg.author.discriminator} used hackban: ${reason}`).then(() => {
+              banMembers.success.push(`\`${user.username}#${user.discriminator}\``)
+              if (banMembers.success.length + banMembers.error.length === idArray.length) {
+                m.edit(`${msg.author.mention}, ${banMembers.success.length === idArray.length ? `${banMembers.success.join(', ')} ${banMembers.success.length > 1 ? 'were' : 'was'} successfully banned.` : `${banMembers.success.length} ${banMembers.success.length > 1 ? 'users were' : 'user was'} banned.\n${banMembers.error.join(', ')} couldn't be banned.`}`)
+              }
+            }).catch(() => {
+              banMembers.error.push(`\`${user.username}#${user.discriminator}\``)
+              if (banMembers.success.length + banMembers.error.length === idArray.length) {
+                m.edit(`${msg.author.mention}, ${banMembers.success.length === idArray.length ? `${banMembers.success.join(', ')} ${banMembers.success.length > 1 ? 'were' : 'was'} successfully banned.` : `${banMembers.success.join(', ')} ${banMembers.success.length > 1 ? 'were' : 'was'} successfully banned.\n${banMembers.error.join(', ')} couldn't be banned.`}`)
+              }
+            })
+          })
+        })
+      })
+    }
+  }
+}
+
+Commands.softban = {
+  name: 'softban',
+  help: 'Softban a user from the server, which kicks the user and deletes their messages.',
```

Comment:
This isn't the most accurate description ever. Softban and kick in the same context doesn't sound good. I'd rather say `Bans and immediately unbans the user`.